core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -108,42 +102,20 @@ public ShuffleExternalSorter(
       int initialSize,
       int numPartitions,
       SparkConf conf,
-      ShuffleWriteMetrics writeMetrics) throws IOException {
+      ShuffleWriteMetrics writeMetrics) {
+    super(memoryManager);
     this.taskMemoryManager = memoryManager;
     this.blockManager = blockManager;
     this.taskContext = taskContext;
-    this.initialSize = initialSize;
     this.peakMemoryUsedBytes = initialSize;
```

Comment:
I think that this is a bug in the old code as well, but I'm pretty sure that this initial `peakMemoryUsedBytes` is wrong: `initialiSize` should actually be called `initialNumElements` or `initialCapacity` or something, since it's measured in terms of number of entires, not bytes.